### PR TITLE
Migrate auto complete

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
Worked on the comments of PR.

Also, Migrated the auto-complete selection to search results in Add tag Modal

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx [hunk 1]
```diff
@@ -10,15 +10,24 @@ import TabToolbar from '../../../components/styled/TabToolbar';
...
-    const description = entityData?.editableProperties?.description || entityData?.properties?.description || '';
+    const localStorageDictionary = localStorage.getItem(EDITED_DESCRIPTIONS_CACHE_NAME);
+    const editedDescriptions = (localStorageDictionary && JSON.parse(localStorageDictionary)) || {};
+    const description = editedDescriptions.hasOwnProperty(urn)
+        ? editedDescriptions[urn]
+        : entityData?.editableProperties?.description || entityData?.properties?.description || '';
+
     const [updatedDescription, setUpdatedDescription] = useState(description);
+    const [isDescriptionUpdated, setIsDescriptionUpdated] = useState(editedDescriptions.hasOwnProperty(urn));
+    const [visible, setVisible] = useState(false);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:30]: let's name this - cancelModalVisible

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:30]: and setCancelModalVisible

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx [hunk 2]
```diff
@@ -65,22 +81,68 @@ export const DescriptionEditor = ({ onComplete }: { onComplete?: () => void }) =
...
+        let delayDebounceFn: ReturnType<typeof setTimeout>;
+        const editedDescriptionsLocal = (localStorageDictionary && JSON.parse(localStorageDictionary)) || {};
+
+        if (isDescriptionUpdated) {
+            delayDebounceFn = setTimeout(() => {
+                editedDescriptionsLocal[urn] = updatedDescription;
+                localStorage.setItem(EDITED_DESCRIPTIONS_CACHE_NAME, JSON.stringify(editedDescriptionsLocal));
+            }, 5000);
+        }
+        return () => clearTimeout(delayDebounceFn);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:105]: Why do we return this function?

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DiscardDescriptionModal.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Modal, Button } from 'antd';
+
+type Props = {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DiscardDescriptionModal.tsx:4]: This looks great!

File: datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx [hunk 4]
```diff
@@ -74,49 +68,56 @@ export default function AddTagTermModal({
...
     const [addTagMutation] = useAddTagMutation();
     const [addTermMutation] = useAddTermMutation();
+    const [tagSearch, { data: tagSearchData }] = useGetSearchResultsLazyQuery();
+    const tagSearchResults = tagSearchData?.search?.searchResults || [];
 
-    const autoComplete = (query: string) => {
-        if (query && query !== '') {
-            getAutoCompleteResults({
+    const handleSearch = (text: string) => {
+        if (text.length > 2) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx:82]: On thinking about this again.. some people have tags and terms of length = 0

Can we update this to be text.length > 0?

File: datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx [hunk 5]
```diff
@@ -74,49 +68,56 @@ export default function AddTagTermModal({
...
-        suggestionsData?.autoCompleteForMultiple?.suggestions.flatMap((entity: AutoCompleteResultForEntity) =>
-            entity.suggestions.map((suggestion: string) =>
-                renderItem(suggestion, entityRegistry.getIcon(entity.type, 14, IconStyleType.TAB_VIEW), entity.type),
-            ),
-        ) || [];
+    const renderSearchResult = (result: SearchResult) => {
+        const displayName = entityRegistry.getDisplayName(result.entity.type, result.entity);
+        const item = renderItem(
+            displayName,
+            entityRegistry.getIcon(result.entity.type, 14, IconStyleType.TAB_VIEW),
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx:100]: We can use IconStyleType.ACCENT here!

File: datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx [hunk 6]
```diff
@@ -74,49 +68,56 @@ export default function AddTagTermModal({
...
+    });
 
-    const autocompleteOptions =
-        options.map((option) => (
-            <Select.Option value={`${option.value}${NAME_TYPE_SEPARATOR}${option.type}`} key={option.value}>
-                {option.label}
-            </Select.Option>
-        )) || [];
+    const inputExistsInTagSearch = tagSearchResults.some((result: SearchResult) => {
+        const displayName = entityRegistry.getDisplayName(result.entity.type, result.entity);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx:115]: This is great!

**Review by jjoyce0510 — COMMENTED** [review]
Minor comments - Otherwise looks great

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:30]: Fixed and commited.

File: datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx (hunk 4 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx:82]: Fixed and commited.

File: datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx (hunk 5 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/shared/tags/AddTagTermModal.tsx:100]: Fixed and commited.

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx (hunk 2 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:105]: Using here to clean-up / unsubscribed the effect.

**Pull request closed without merging**